feat: add sub fields for address

omarNaifer12 opened the pull request:
https://github.com/user-attachments/assets/8fe9079d-9b66-4b42-8b11-ad713e9de666

FelixMalfait commented on packages/twenty-front/src/utils/displaySubFieldsAddress.ts:
```diff
@@ -0,0 +1,30 @@
+} from 'twenty-shared/src/types/AddressFieldsType';
+import { isDefined } from 'twenty-shared/utils';
+
+export const extractSubFieldsAddressValues = (
```
1 util per file + could be named formatAddressDisplay / joinAddressFieldValues for example

FelixMalfait commented on packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/AddressFieldDisplay.tsx:
```diff
@@ -1,19 +1,14 @@
 
-  return <TextDisplay text={content} />;
+  const parsedFieldValue = extractSubFieldsAddressValues(fieldValue, subFields);
+  return <TextDisplay text={parsedFieldValue} />;
```
Maybe the PR is still ongoing and purposely not finished, but I think we want to change not just the display value as you did (great start) but also the values being display in edit mode

omarNaifer12 replied:
Did you mean we should also block the user from typing in edit mode if the input field isn’t selected in the subfields?

FelixMalfait replied:
Yes ideally we should do even more than prevent from typing but instead not display the fields at all (adapt as elengantly as it’s possible without complexifying too much)

FelixMalfait commented on packages/twenty-front/src/modules/settings/data-model/fields/forms/address/constants/DefaultSelectionAddressWithMessages.ts:
```diff
@@ -0,0 +1,32 @@
+import { msg } from '@lingui/core/macro';
+import { AllowedAddressSubField } from 'twenty-shared/src/types/AddressFieldsType';
+
+export const DefaultSelectionAddressWithMessages: {
```
casing looks off. Should be ALL_CAPS

FelixMalfait reviewed (commented):
Great work again. Made a few minor comments. Biggest missing part is to update the edit form to adapt to selected subfields

omarNaifer12 commented:
Hey @FelixMalfait, I’ve addressed your feedback and handled the edit mode, but some checks are still failing. The logs aren’t clear and don’t show where the errors happen so I’m not sure why or how to resolve them. Let me know if there’s anything else.

FelixMalfait reviewed (approved):
Great! Thanks a lot!
